pkg/nfd-worker/nfd-worker.go

```diff
@@ -238,7 +263,12 @@ func (w *nfdWorker) Run() error {
 
 			// Always re-label after a re-config event. This way the new config
 			// comes into effect even if the sleep interval is long (or infinite)
-			labelTrigger = time.After(0)
+			labelTrigger.Stop()
+			labelTrigger = w.createTicker()
+			err = w.runFeatureDiscovery()
```

Comment:
Wouldn't it be simpler just to run `w.runFeatureDiscovery` here instead of reinitializing  the Ticker?

Reply:
`core.sleepInterval` can change if the config changes. I restart the Ticker for the possible new interval

Reply:
It makes sense, I haven't thought about this.